### PR TITLE
syncrepl changes

### PR DESCRIPTION
**Pull request by Firstyear**

### CODE REVIEW

File: cli/src/ldap.rs [hunk 1]
```diff
@@ -228,18 +228,26 @@ async fn main() {
                         }
                         println!("");
                     }
-                    for entry_uuid in &delete_uuids {
-                        println!("delete entryuuid: {}", entry_uuid);
-                    }
-                    if !present_uuids.is_empty() {
-                        println!("");
+                    if let Some(d_uuids) = &delete_uuids {
```
Comment (nitnelave) [inline cli/src/ldap.rs:231]: `for entry_uuid in delete_uuids.as_ref().unwrap_or_default()` ?

Comment (Firstyear) [reply cli/src/ldap.rs:231]: That's a neater solution for certain :)

Comment (Firstyear) [reply cli/src/ldap.rs:231]: Also sorry for the yolo merge, I needed to put this in for a PR for kani that's quite important RN.

Comment (nitnelave) [reply cli/src/ldap.rs:231]: Ah, no problem, do what you must, I'm just drive-by commenting :)

File: client/src/syncrepl.rs [hunk 2]
```diff
@@ -133,9 +133,11 @@ impl LdapClient {
                     // state of the content after performing the synchronization of the
                     // entries in the set.
                     if refresh_deletes {
-                        delete_uuids.extend(syncuuids.into_iter())
+                        let d_uuids = delete_uuids.get_or_insert_with(Vec::default);
```
Comment (nitnelave) [inline client/src/syncrepl.rs:136]: `get_or_insert_default()` is coming, still unstable for now

File: cli/src/ldap.rs (hunk 1 above)
Comment (Firstyear) [reply cli/src/ldap.rs:231]: They're good comments.

File: client/src/syncrepl.rs (hunk 2 above)
Comment (Firstyear) [reply client/src/syncrepl.rs:136]: Yeah, I know. I found the docs for that and was sad I can't use it yet.

**Pull request merged**